R/family.R

```diff
@@ -10,21 +10,23 @@ poisson_nonprobsvy <- function(link = "log") {
   structure(list(mu = mu,
                  variance = variance,
                  mu_der = mu_der,
-                 residuals = residuals),
+                 residuals = residuals,
+                 family = "poisson"),
             class = "family")
 }
 
 gaussian_nonprobsvy <- function(link = "identity") {
   mu <- function(eta) eta
-  variance <- function(mu, y = NULL) rep.int(1, length(mu)) #mean((y - mu)^2) rep.int(1, length(mu))
+  variance <- function(mu, y = NULL) mean((y - mu)^2) #rep.int(1, length(mu)) rep.int(1, length(mu))
```

Comment:
Just as a reminder you don't have to write all those for every exponential family you want to use you can also just do something simpler like:
```r
gaussian_nonprobsvy <- function(link = "identity") {
  x <- stats::gaussian(link = link)
  x$mu_der <- function(mu) 1
  # Sooner or later you will also probably need a class for those as well
  class(x) <- c(class(x), "nonprobsvy_family")
}
```